django/core/files/locks.py

```diff
@@ -107,9 +107,15 @@ def unlock(f):
             return True
     else:
         def lock(f, flags):
-            ret = fcntl.flock(_fd(f), flags)
-            return ret == 0
+            try:
+                fcntl.flock(_fd(f), flags)
+                return True
+            except OSError:
+                return False
 
         def unlock(f):
-            ret = fcntl.flock(_fd(f), fcntl.LOCK_UN)
-            return ret == 0
+            try:
+                fcntl.flock(_fd(f), fcntl.LOCK_UN)
+                return True
+            except OSError:
+                return False
```

Comment:
TBH I'm not sure about suppressing `OSError`s in `lock()`/`unlock()` because it's backward incompatible (even if both methods are undocumented) and can cause many issues if folks rely on the current behavior :thinking: We will also suppress errors related with bad descriptors and flags which is not desirable. Maybe we should simply return `True` if it doesn't fail (\cc @jamercee):
```python
def lock(f, flags):
    fcntl.flock(_fd(f), flags)
    return True

def unlock(f, flags):
    fcntl.flock(_fd(f), fcntl.LOCK_UN)
    return True
```

or catch `BlockingIOError` only if `lock()` (because `unlock()` can raise an exception only on bad values):

```python
def lock(f, flags):
    try:
        fcntl.flock(_fd(f), flags)
        return True
    except BlockingIOError:
        return False

def unlock(f, flags):
    fcntl.flock(_fd(f), fcntl.LOCK_UN)
    return True
```
Release notes are required for the second option.

Reply:
IMO, The first option:
> Maybe we should simply return True if it doesn't fail 

Would be better.

Reply:
maybe we could do the first option here https://code.djangoproject.com/ticket/32708#ticket ?

 it seems to be the best option as it doesn't break third party code relying on the OSError being raised